packages/extensionmanager/src/widget.tsx

```diff
@@ -282,107 +285,129 @@ function ListEntry(props: ListEntry.IProperties): React.ReactElement<any> {
                 <Button
-                  onClick={() => props.performAction('disable', entry)}
+                  onClick={() => props.performAction('install', entry)}
                   minimal
                   small
                 >
-                  {trans.__('Disable')}
+                  {trans.__('Update')}
                 </Button>
               )}
-              {entry.installed && !entry.enabled && (
+            {entry.installed && entry.pkg_type == 'source' && (
+              <Button
+                onClick={() => props.performAction('uninstall', entry)}
+                minimal
+                small
+              >
+                {trans.__('Uninstall')}
+              </Button>
+            )}
+            {entry.enabled && entry.pkg_type == 'source' && (
+              <Button
+                onClick={() => props.performAction('disable', entry)}
+                minimal
+                small
+              >
+                {trans.__('Disable')}
+              </Button>
+            )}
+            {entry.installed && entry.pkg_type == 'source' && !entry.enabled && (
+              <Button
+                onClick={() => props.performAction('enable', entry)}
+                minimal
+                small
+              >
+                {trans.__('Enable')}
+              </Button>
+            )}
+            {entry.installed && entry.pkg_type == 'prebuilt' && (
+              <div className="jp-extensionmanager-entry-buttons">
                 <Button
-                  onClick={() => props.performAction('enable', entry)}
+                  onClick={() =>
+                    showDialog({
+                      title,
+                      body: (
+                        <div>
+                          {getprebuiltUninstallInstruction(entry, trans)}
+                        </div>
+                      ),
+                      buttons: [
+                        Dialog.okButton({
+                          label: trans.__('OK'),
+                          caption: trans.__('OK')
+                        })
+                      ]
+                    }).then(result => {
+                      return result.button.accept;
+                    })
+                  }
                   minimal
                   small
                 >
-                  {trans.__('Enable')}
+                  {trans.__('About')}
                 </Button>
-              )}
-            </div>
-          )}
+              </div>
+            )}
+          </div>
         </div>
       </div>
     </li>
   );
 }
 
+function getprebuiltUninstallInstruction(
+  entry: IEntry,
+  trans: TranslationBundle
+): JSX.Element {
+  if (entry.install?.uninstallInstructions) {
+    return (
+      <div>
+        <p>
+          {trans.__(`This is a prebuilt extension. To uninstall it, please
+    apply following instructions.`)}
+        </p>
+        <p>{entry.install?.uninstallInstructions}</p>
```

Comment:
Even though it might not be easily possible to localize the current uninstall message (since it contains the name of the package), maybe this blob of text should still be wrapped with a call to `trans.__()`?

Since extension authors control the content of `install.json`, they could decide to have different instructions if they want to. And be able to localize these instructions.

Reply:
I have pushed that change. Thx.